### PR TITLE
complete the interactive form

### PR DESCRIPTION
**Pull request by igakigongo**
**Tasks**
- Handle for submission for creating the books and adding them to the redux store
- Handle removing the book from the rendered list and redux store

### CODE REVIEW

File: src/containers/books-form.js [hunk 1]
```diff
@@ -61,13 +70,24 @@ const BooksForm = () => {
...
 
               <Form.Group controlId="selectListControlId">
                 <Form.Label>Category</Form.Label>
-                <Form.Control as="select" className="border-success" defaultValue={category} onChange={e => { setCategory(e.currentTarget.value); }}>
+                <Form.Control
+                  as="select"
+                  className="border-success"
+                  defaultValue={category}
+                  onChange={e => { setCategory(e.currentTarget.value); }}
+                >
```
Comment (shloch) [inline src/containers/books-form.js:90]: good

File: src/containers/books-form.js [hunk 2]
```diff
@@ -30,28 +35,32 @@ const useInputWithValidation = (initialState, required = false) => {
...
 
+  const titleIsValid = () => titlePristine !== null && titlePristine && !title;
 
   const handleSubmit = e => {
     e.preventDefault();
+    if (titleIsValid()) return;
+    dispatch(createBook(Book(title, category)));
+    titleRef.current.value = '';
+    setCategory(BookCategories.ACTION);
   };
```
Comment (shloch) [inline src/containers/books-form.js:62]: good

File: src/containers/books-list.js [hunk 3]
```diff
@@ -2,43 +2,67 @@ import React from 'react';
...
           </tr>
         </thead>
         <tbody>
-          {books.map(book => (<Book book={book} key={book.id} />))}
+          {books.map(book => (
+            <Book
+              book={book}
+              handleRemoveBook={() => { handleRemoveBook(book); }}
+              key={book.id}
+            />
```
Comment (shloch) [inline src/containers/books-list.js:37]: good

File: src/components/book.js [hunk 4]
```diff
@@ -1,19 +1,28 @@
...
+
   return (
     <tr>
       <th scope="row"><code className="bg-light p-2 shadow-sm rounded">{id}</code></th>
       <td>{title}</td>
       <td>{category}</td>
+      <td className="text-center">
+        <Button onClick={() => { handleRemoveBook(book); }} size="sm" type="button" variant="outline-danger">
+          Remove Book
+        </Button>
```
Comment (shloch) [inline src/components/book.js:17]: good job

**Review by shloch — APPROVED** [review]
## hello @igakigongo 

### it is clear that you know what you're doing, great job

accomplished

File: src/containers/books-form.js (hunk 1 above)
Comment (igakigongo) [reply src/containers/books-form.js:90]: Thanks too for the quick review.

> good

**Pull request merged**